### DISCUSSION TITLE
parser：关于库函数中存在的一个隐式转换语法的疑惑

### DISCUSSION THREAD

**Discussion by PiCaHor** [arcsysu/SYsU-lang#78] [Q&A] [answered]
在sysulib库中如果引用putfloat 与putint ，clang会输出
![1679970427782](https://user-images.githubusercontent.com/74459835/228111392-744e0ecb-a5e1-4e4b-a334-93d6bde8fd9f.png)
对于其中的floatintcast这一步的隐式转换的语法存在困惑，这里传的参数在定义中也是float与int类型，为何这里需要转换成double类型

**Comment by wufeng15226** [comment, marked as answer]
因为此处 putfloat() 被隐式声明，对于未知参数函数的实际参数，float 默认转换为了 double。可以参考 https://stackoverflow.com/questions/9182763/implicit-function-declarations-in-c
